src/Features/Core/Portable/Completion/CompletionServiceWithProviders.cs

```diff
@@ -194,6 +218,12 @@ internal protected CompletionProvider GetProvider(CompletionItem item)
             {
                 lock (_gate)
                 {
+                    if (!_roleProvidersInitialized)
+                    {
+                        CreateRoleProviders(ImmutableHashSet.Create<string>());
```

Comment:
Why is ```ImmutableHashSet.Create<string>()``` the right value to use here?  Can you explain/comment why?

Reply:
Also, still not quite understanding how this won't still have a problem elsewhere. for example, say that the item references a provider that thsi service doesn't know about.  Won't this just return null, causing a crash elsewhere?  or are we resilient to that?

Reply:
Can this be answered?